atmo/jobs/models.py

```diff
@@ -120,13 +112,15 @@ def has_never_run(self):
         Looks at both the cluster status and our own record when
         we asked it to run.
         """
-        return (self.most_recent_status == self.DEFAULT_STATUS or
-                self.last_run_date is None)
+        return (self.latest_run is None or
+                self.latest_run.status == DEFAULT_STATUS or
+                self.latest_run.run_date is None)
```

Comment:
I tried the `.exists()` query but since `latest_run` is cached on the model doing these few checks is easier for testing vs having to save to the db. I think it's also more clear than a query and matches the intention of the method better.

I could easily make `DEFAULT_STATUS = 'CREATED'` if you'd like.